src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -597,6 +597,15 @@ EnsureSequenceTypeSupported(Oid seqOid, Oid attributeTypeId, Oid ownerRelationId
 		forboth_int_oid(currentAttnum, attnumList, currentSeqOid,
 						dependentSequenceList)
 		{
+			if (currentAttnum == InvalidAttrNumber)
+			{
+				/*
+				 * InvalidAttrNumber indicates that the column does not
+				 * have default nextval(), so no need to check.
```

Comment:
is the refobjsubid in pg_depend 0 in that case?

Reply:
It is >0 actually. I think putting  `InvalidAttrNumber` in `GetDependentSequencesWithRelation` is the source of the confusion. I also don't like that much, given that it also confused you seems like a good sign to improve.  Let me make this code flow more explicit

```SQL
CREATE SEQUENCE "unrelated_sequence";
CREATE TABLE test(a int, b int);
ALTER SEQUENCE "unrelated_sequence" OWNED BY test.b;


SELECT pg_identify_object_as_address(classid, objid, objsubid), pg_identify_object_as_address(refclassid, refobjid, refobjsubid), refobjsubid FROM pg_depend WHERE refobjid = 'test'::regclass;

-7498=# SELECT pg_identify_object_as_address(classid, objid, objsubid), pg_identify_object_as_address(refclassid, refobjid, refobjsubid), refobjsubid FROM pg_depend WHERE refobjid = 'test'::regclass;
┌─[ RECORD 1 ]──────────────────┬─────────────────────────────────────────────┐
│ pg_identify_object_as_address │ (type,{public.test},{})                     │
│ pg_identify_object_as_address │ (table,"{public,test}",{})                  │
│ refobjsubid                   │ 0                                           │
├─[ RECORD 2 ]──────────────────┼─────────────────────────────────────────────┤
│ pg_identify_object_as_address │ (sequence,"{public,unrelated_sequence}",{}) │
│ pg_identify_object_as_address │ ("table column","{public,test,b}",{})       │
│ refobjsubid                   │ 2                                           │
└───────────────────────────────┴─────────────────────────────────────────────┘

Time: 1.336 ms
```

Reply:
@marcocitus pushed a revision. Is this easier to follow for you?